pkg/nfd-master/node-updater-pool.go

```diff
@@ -0,0 +1,74 @@
+package nfdmaster
+
+import (
+	"sync"
+
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type nodeUpdaterPool struct {
+	queue     workqueue.RateLimitingInterface
+	wg        sync.WaitGroup
+	nfdMaster nfdMaster
+}
+
+var lock = &sync.Mutex{}
+var instance *nodeUpdaterPool
+
+func getNodeUpdaterPool(nfdMaster nfdMaster) *nodeUpdaterPool {
```

Comment:
This singleton approach looks a bit weird as you give nfdMaster as an argument but that argument is totally omitted in subsequent calls.

How about changing this to

```suggestion
func newNodeUpdaterPool(nfdMaster nfdMaster) *nodeUpdaterPool
```

and then calling this in NewNfdMaster and storing the instance in the nfdMaster struct?

Reply:
I agree, It would be much more readable, since the nodeUpdaterPool is only used from within `nfdMaster`.

Reply:
Wouldn't that result in a circular dependency?

Reply:
`nfdMaster contains nodeUpdaterPool` and `nodeUpdaterPool contains nfdMaster`.

Reply:
Think about it like doubly-linked list